Documentation/Debugger/gdb/Windows Subsystem for Linux.md

```diff
@@ -0,0 +1,88 @@
+# Windows 10's Windows Subsystem for Linux
+With the release of Windows 10 Creator's Update, you will now be able to use Visual Studio Code and the Microsoft C/C++ extension to debug your `Windows Subsystem for Linux (WSL)` Bash on Ubuntu projects.
+
+Code can be written on Windows itself using VSCode and debugged through `bash.exe` to the Bash on Windows layer. 
+
+**NOTE: Creator's Update is required due to bugfixes within the subsystem that we rely on to provide debugging. Debugging using a previous version of WSL is unsupported and likely will not work.**
+
+## Prerequisites
+* [Windows 10 Creator's Update with Windows Subsystem for Linux and Bash.](https://msdn.microsoft.com/en-us/commandline/wsl/install_guide)
+* Install g++/gcc and gdb to allow compiling and debugging.
+* [Visual Studio Code](https://code.visualstudio.com) + Microsoft C/C++ extension for VSCode.
+
+## How-To
+To debug, commands will be routed from Windows through `bash.exe` to setup debugging. Because our extension runs as a 32-bit process, it will need to use the `C:\Windows\SysNative` folder to access the `bash.exe` executable that is normally in `C:\Windows\System32`. We will be using the `"pipeTransport"` ability within the extension to do debugging and `"sourceFileMap"` to map the source from the subsystem's paths back to Windows path. 
```

Comment:
`setup` is used as a verb a few times in this document, but when spelled as one word it is a noun.  Replace with `set up` (two words)